x-pack/plugin/sql/qa/src/main/java/org/elasticsearch/xpack/sql/qa/jdbc/ResultSetTestCase.java

```diff
@@ -791,7 +789,188 @@ public void testGettingInvalidFloat() throws Exception {
                     sqle.getMessage());
         });
     }
-    
+
+    //
+    // BigDecimal fetching testing
+    //
+    static final Map<Class<? extends Number>, Integer> JAVA_TO_SQL_NUMERIC_TYPES_MAP = new HashMap<>() {{
+        put(Byte.class, Types.TINYINT);
+        put(Short.class, Types.SMALLINT);
+        put(Integer.class, Types.INTEGER);
+        put(Long.class, Types.BIGINT);
+        put(Float.class, Types.REAL);
+        put(Double.class, Types.DOUBLE);
+        // TODO: no half & scaled float testing
```

Comment:
Why not adding those too?

Reply:
It probably should be done indeed, but I was thinking to keep it focused on BigDecimal for now, since adding coverage for half- and scaled- floats would require extending the testing for the rest of the types too.